BricksAndHearts.UnitTests/ServiceTests/Property/PropertyServiceTests.cs

```diff
@@ -402,4 +410,74 @@ public void CreateNewPublicViewLink_UpdateDatabase()
+    {
+        // Arrange
+        var logger = A.Fake<ILogger<PostcodeApiService>>();
+        var messageHandler = A.Fake<HttpMessageHandler>();
+        const string postcode = "eeeeee";
+        var responseBody = string.Empty;
+        var response = new HttpResponseMessage
+        {
+            StatusCode = HttpStatusCode.NotFound,
+            Content = new StringContent(responseBody)
+        };
+        // Slightly icky because "SendAsync" is protected
+        A.CallTo(messageHandler).Where(c => c.Method.Name == "SendAsync").WithReturnType<Task<HttpResponseMessage>>().Returns(response);
+        var httpClient = new HttpClient(messageHandler);
+        var postcodeApiService = new PostcodeApiService(logger, null!, httpClient);
+        var service = new PropertyService(null!, postcodeApiService);
+
+        // Act
+        var result = await service.SortPropertiesByLocation(postcode);
+        
+        // Assert
+        result.Should().BeNull();
+    }
+
+    [Fact]
+    public async void SortPropertiesByLocation_WhenCalledWithValidPostcode_ReturnPropertiesSorted()
+    {
+        // Arrange
+        await using var context = Fixture.CreateReadContext();
+        var logger = A.Fake<ILogger<PostcodeApiService>>();
+        var messageHandler = A.Fake<HttpMessageHandler>();
+        const string postcode = "eh11ad";
+        var responseBody = await File.ReadAllTextAsync($"{AppDomain.CurrentDomain.BaseDirectory}/../../../ServiceTests/Api/PostcodeioApiResponse.json");
+        var response = new HttpResponseMessage
+        {
+            StatusCode = HttpStatusCode.OK,
+            Content = new StringContent(responseBody)
+        };
+        // Slightly icky because "SendAsync" is protected
+        A.CallTo(messageHandler).Where(c => c.Method.Name == "SendAsync").WithReturnType<Task<HttpResponseMessage>>().Returns(response);
+        var httpClient = new HttpClient(messageHandler);
+        var postcodeApiService = new PostcodeApiService(logger, context, httpClient);
+        var service = new PropertyService(context, postcodeApiService);
+        
+        
+        // Act
+        var result = await service.SortPropertiesByLocation(postcode);
+        
+        // Assert
+        var correctList = new List<PropertyDbModel>
+        {
+            context.Properties.Single(p => p.TownOrCity == "Leeds"),
+            context.Properties.Single(p => p.TownOrCity == "Peterborough"),
+            context.Properties.Single(p => p.TownOrCity == "London"),
+            context.Properties.Single(p => p.TownOrCity == "Brighton"),
+        };
+        var propertyDbModelList = result!.FindAll(p => p.LandlordId == 3);
+        propertyDbModelList.Should().NotBeNullOrEmpty();
+        propertyDbModelList.Should().HaveCount(4);
+        var i = 0;
+        while (i < 4)
+        {
+            propertyDbModelList[i].Should().BeEquivalentTo(correctList[i]);
+            i++;
+        }

```

Comment:
Can you just do `propertyDbModelList.Should().BeEquivalentTo(correctList);`, or does that not work?

Reply:
I think that the list isn't ordered... as in, it seems to think that [1,2,3,4] is the same as [1,2,4,3] - I might be wrong tho